unit/deserialize.go

```diff
@@ -43,36 +43,83 @@ var (
 	ErrLineTooLong = fmt.Errorf("line too long (max %d bytes)", SYSTEMD_LINE_MAX)
 )
 
-// Deserialize parses a systemd unit file into a list of UnitOption objects.
+// Deserialize parses a systemd unit file into a list of UnitOptions.
 func Deserialize(f io.Reader) (opts []*UnitOption, err error) {
-	lexer, optchan, errchan := newLexer(f)
+	_, options, err := DeserializeAll(f)
+	return options, err
+}
+
+// DeserializeSections deserializes into a list of UnitSections.
+func DeserializeSections(f io.Reader) ([]*UnitSection, error) {
+	sections, _, err := DeserializeAll(f)
+	return sections, err
+}
+
+// DeserializeAll deserializes into UnitSections and UnitOptions.
+func DeserializeAll(f io.Reader) ([]*UnitSection, []*UnitOption, error) {
```

Comment:
I am having a bit of an hard time understanding how you are supposed to use this API.
Are the two slice supposed to contain almost the same information, but in a different format (sections vs options)? What's the intended usecase for this?

Reply:
I added a separate data structure as the original one did not fully support needed information. I added a new one rather than modified the existing one to support backwards compatibility. The new structure *does* contain *almost* the same information but also includes a bit of information that was not possible to express in the original structure, namely which options are in which sections when there are duplicate section names. 

I'm using this to support reading and writing `[Route]` sections into network interface unit files. The assumption of the current unit code seems to be that `section` names are unique. This is not true for `Route` sections as there can be multiple `Route` sections in a single file.

Reply:
Thanks. I do understand the underlying logic and goals. My doubts were about:
 * is there a usecase where you want to deserialize both in a single call? If not, this can be a private function.
 * balancing backward compatibility vs API usability, would it be better to just have `DeserializeOptions` and `DeserializeSections`?

Reply:
Sorry, I misunderstood your comment entirely. Yes, this is just a helper function and I can think of no time you'd want both UnitSections and UnitOptions at once. This is just a utility function for  `Deserialize` and `DeserializeSections`. I will make this private. 

Are you suggesting renaming `Deserialize` to `DeserializeOptions`? Or writing a wrapper function and keeping `Deserialize`? Either way is OK with me, but keeping `Deserialize` would minimize disruption. Let me know which you prefer.

Reply:
> Are you suggesting renaming Deserialize to DeserializeOptions?

I was thinking about this, yes. Not directly suggesting as I was not really sure whether I prefer compatibility over clarity.

I think in the end the best approach would be having:
 1) `DeserializeSections`
 2) `DeserializeOptions`
 3) `Deserialize` which forwards to `DeserializeOptions` and a deprecation notice in its docstring. Something like `this will be dropped in the future, use DeserializeOptions instead`.